### PR TITLE
docs(book): start documenting ExExes

### PR DESCRIPTION
**Pull request by shekhirin**
The easiest way to review this is to run `mdbook serve` locally on this PR and explore the book in your browser.

### CODE REVIEW

File: book/developers/exex/exex.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Execution Extensions (ExEx)
+
+## What are Execution Extensions?
+
+Execution Extensions allow developers to build their own infrastructure that relies on Reth
+as a base for driving the chain (be it [Ethereum](../../run/mainnet.md) or [OP Stack](../../run/optimism.md)) forward.
+
+Execution Extension is a task that derives its state from Reth's state.
```
Comment (Rjected) [inline book/developers/exex/exex.md:8]: should this be "changes in reth's state"?

File: book/developers/exex/exex.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+Some examples of such state derives are rollups, bridges, and indexers.
```
Comment (Rjected) [inline book/developers/exex/exex.md:9]:
```suggestion
Some examples of such state derivations are rollups, bridges, and indexers.
```

File: book/developers/exex/hello-world.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+It's the only way to communicate to the main node that the ExEx has finished processing the specified height
+and it's safe to prune the associated data.
+
+</div>
+
+What we've arrived at is the [minimal ExEx example](https://github.com/paradigmxyz/reth/blob/b8cd7be6c92a71aea5341cdeba685f124c6de540/examples/exex/minimal/src/main.rs) that we provide in the Reth repository.
+
+## What's next?
+
+Let's do something a bit more interesting, and see how you can keep track of some state inside your ExEx.
```
Comment (Rjected) [inline book/developers/exex/hello-world.md:176]: probably would remove this until there is a following section with the stateful exex

File: book/developers/exex/how-it-works.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+- Shutting ExExes down when the node is shut down
+
+## Pruning
+
+Pruning deserves a special mention here.
+
+ExExes **SHOULD** emit an [`ExExEvent::FinishedHeight`](https://reth.rs/docs/reth_exex/enum.ExExEvent.html#variant.FinishedHeight)
+event to signify what blocks have been processed. This event is used by Reth to determine what state can be pruned.
+
+An ExEx will only receive notifications for blocks greater than the block emitted in the event.
```
Comment (Rjected) [inline book/developers/exex/how-it-works.md:24]:
```suggestion
An ExEx will only receive notifications for block numbers greater than the block in the most recently emitted `FinishedHeight` event.
```

**Review by Rjected — COMMENTED** [review]
looks pretty good! mostly minor comments

**Review by onbjerg — COMMENTED** [review]
some minor things, looks great!

File: book/developers/exex/hello-world.md [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+) -> eyre::Result<()> {
+    loop {}
+}
+
+fn main() -> eyre::Result<()> {
+    reth::cli::Cli::parse_args().run(|builder, _| async move {
+        let handle = builder
+            .node(EthereumNode::default())
+            .install_exex(
+                "Hello World",
```
[the remark this answers is not in the document]
Comment (shekhirin) [reply book/developers/exex/hello-world.md]: yep, turned it into a `my-exex`

**Review by onbjerg — APPROVED** [review]

**Review by mattsse — APPROVED** [review]

File: book/developers/exex/hello-world.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+    reth::cli::Cli::parse_args().run(|builder, _| async move {
+        let handle = builder
+            .node(EthereumNode::default())
+            .install_exex("my-exex", |ctx| async move { Ok(my_exex(ctx)) })
+            .launch()
+            .await?;
+
+        handle.wait_for_node_exit().await
+    })
+}
```
[the remark this answers is not in the document]
Comment (shekhirin) [reply book/developers/exex/hello-world.md:83]: unfortunately, there seems to be no way to compile the snippets that have any dependencies :/

**Pull request merged**